core/linker/default.h

```diff
@@ -36,17 +36,52 @@ ENTRY(main_entry)
 #error "uVisor does not fit into the target memory. UVISOR_SRAM_LENGTH_USED must be smaller than UVISOR_SRAM_LENGTH_AVAIL."
 #endif /* UVISOR_SRAM_LENGTH_USED > UVISOR_SRAM_LENGTH_AVAIL */
 
+
+#ifdef ARCH_MPU_ARMv7M
+/* uVisor stack on MPU_ARMv7M MPU architecture:
+*
+* .---------------------.    --
+* |        Guard        |       \
+* |  [STACK_GUARD_BAND] |       |
+* +---------------------+       |
+* |                     |       |
+* |        Stack        |        } => [TOTAL_STACK_SIZE]
+* |     [STACK_SIZE]    |       |
+* |                     |       |
+* +---------------------+       |
+* |        Guard        |       |
+* |  [STACK_GUARD_BAND] |       /
+* '---------------------'    --
+*/
+#define TOTAL_STACK_SIZE 2048   /* Must be a power of 2 */
+/* Default uVisor own stack guard band
+ * Note: Currently we use the stack guard to isolate the uVisor stack on
+ *       MPU_ARMv7M MPU architecture only.
+ *       In order to use SRD (sub region disabled) we set the guard to
+ *       be 1/8th of the total stack size */
+#define STACK_GUARD_BAND (TOTAL_STACK_SIZE >> 3)
+/* Default uVisor stack size
+ * Note: This is uVisor own stack, not the boxes' stack. */
+#define STACK_SIZE (TOTAL_STACK_SIZE - STACK_GUARD_BAND - STACK_GUARD_BAND)
+#else
 /* Default uVisor stack size
  * Note: This is uVisor own stack, not the boxes' stack. */
-#if !defined(STACK_SIZE)
```

Comment:
If STACK_SIZE is defined then it is overriden now.

Reply:
AFAIU since STACK_SIZE can come only from uVisor core (and not from application), it is only the one who builds uVisor that can define the stack size.
Therefore I think it should be defined  only in the linker script so I have eliminated the "if !define..." line

Reply:
We still need to keep the ifdef here. See docs/core/PORTING.md. We can provide a default stack size here, but targets need to be able to override this value.

Reply:
From your mail I understood that we can it without the #ifdef.
Did I misunderstand you?

Reply:
I believe that not allowing to pre-define STACK_SIZE is the best solution.
Nevertheless, I added the #ifdef STACK_SIZE as you suggested.